libraries/core-react/src/hooks/useOutsideClick.ts

```diff
@@ -2,10 +2,15 @@ import { useEffect, MutableRefObject } from 'react'
 
 export const useOutsideClick = (
   ref: MutableRefObject<HTMLElement>,
+  anchorEl: HTMLElement,
   callback: () => void,
 ): void => {
   const handleClick = (e: MouseEvent) => {
-    if (ref.current && !ref.current.contains(e.target as Node)) {
+    if (
+      ref.current &&
+      !ref.current.contains(e.target as Node) &&
+      !anchorEl.contains(e.target as Node)
```

Comment:
` !anchorEl.contains(e.target as Node)` was added to avoid close on first render

Reply:
This seems as a weird behaviour... What is causing this, because we aren't clicking anywhere on first render...